pkg/didcomm/protocol/exchange/exchange.go

```diff
@@ -30,11 +75,56 @@ func GenerateInviteWithPublicDID(invite *Invitation) (string, error) {
 	return encodedExchangeInvitation(invite)
 }
 
+// GenerateInviteWithKeyAndEndpoint generates the DID exchange invitation string with recipient key and endpoint
+func GenerateInviteWithKeyAndEndpoint(invite *Invitation) (string, error) {
+	if invite.ID == "" || invite.ServiceEndpoint == "" || len(invite.RecipientKeys) == 0 {
+		return "", errors.New("ID, Service Endpoint and Recipient Key are mandatory")
+	}
+
+	return encodedExchangeInvitation(invite)
```

Comment:
@rolsonquadras you shouldn't assume the encoding for an invitation to exchange

Reply:
@llorllale spec define the encoding for invitation

`invitation_string = b64urlencode(<invitation_message>)`

https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange#0-invitation-to-exchange